File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    maxW=-1
    maxH=-1
    for s in sizes:
        if s[0] < s[1]:
            s[0],s[1]=s[1],s[0]
        
    for s in sizes:
        if s[0]>maxW:
            maxW=s[0]
        if s[1]>maxH:
            maxH=s[1]
    return maxW*maxH

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let maxW = 0;
    let maxH = 0;

    for (let s of sizes) {
        if (s[0] < s[1]) [s[0], s[1]] = [s[1], s[0]]
        if (s[0] > maxW) maxW = s[0]
        if (s[1] > maxH) maxH = s[1]
    }
    
    return maxW * maxH;
}